taskcat/testing/base_test.py

```diff
@@ -0,0 +1,135 @@
+import uuid
+from pathlib import Path
+from typing import Any, Dict
+
+from taskcat._cli_core import GLOBAL_ARGS
+from taskcat._config import Config
+
+from .ab_test import Test
+
+
+class BaseTest(Test):
+    """A Generic Test Class that implements the passed
+    and uid properties. Any subclass will still need to
+    implement the the run and clean_up methods.
+    """
+
+    def __init__(self, config: Config):
+        self.config: Config = config
+        self.passed: bool = False
+        self.result: Any = None
+
+    @property
+    def config(self) -> Config:
+        return self._config
+
+    @config.setter
+    def config(self, config: Config) -> None:
+        # It should be possible to check if config is already set
+        # and if it is throw an exception. Might be needed since
+        # child objects rely on the configs uid.
+        self._config = config
+
+    @property
+    def passed(self) -> bool:
+        return self._passed
+
+    @passed.setter
+    def passed(self, new_value: bool) -> None:
+        self._passed = new_value
+
+    @property
+    def result(self) -> Any:
+        return self._result
+
+    @result.setter
+    def result(self, new_value: Any) -> None:
+        self._result = new_value
+
+    def __enter__(self):
+
+        try:
+            self.run()
+        except BaseException as ex:
+            self.clean_up()
+            raise ex
+
+        return self.result
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        # we could optionally call self.report() on exiting.
+        self.clean_up()
+
+    @classmethod
+    def from_file(
+        cls,
+        project_root: str = "./",
+        input_file: str = "./.taskcat.yml",
+        regions: str = "ALL",
+        enable_sig_v2: bool = False,
+    ):
```

Comment:
I gave every test a `from_file` and `from_dict` classmethod, but I think in the future this would be best on the Config object its self.